src/server/courseInstance/courseInstance.service.ts

```diff
@@ -322,4 +327,26 @@ export class CourseInstanceService {
       offered, preEnrollment, studyCardEnrollment, actualEnrollment,
     };
   }
+
+  public async getRoomSchedule(
+    roomId: string,
+    term: TERM,
+    calendarYear: string
+  ): Promise<RoomScheduleResponseDTO[]> {
+    return this.roomScheduleRepository
+      .createQueryBuilder('block')
+      .leftJoinAndMapMany(
+        'block.faculty',
+        FacultyListingView,
+        'instructors',
+        'instructors."courseInstanceId" = block."courseInstanceId"'
+      )
+      .where('block."roomId" = :roomId', { roomId })
+      .andWhere('block.term = :term', { term })
+      .andWhere('block."calendarYear" = :calendarYear', { calendarYear })
+      .orderBy('weekday', 'ASC')
+      .addOrderBy('"startHour"', 'ASC')
+      .addOrderBy('"startMinute"', 'ASC')
```

Comment:
This also needs `.addOrderBy('instructors."instructorOrder"', 'ASC')` (as on line 137) to make sure that the instructors end up being listed in the correct order in the schedule.